src/error-handling.js

```diff
@@ -5,7 +5,21 @@ const fs = require('fs-extra');
 async function validatePathName (dir) {
   try {
     // eslint-disable-next-line security/detect-non-literal-fs-filename
-    const files = await fs.readdir(dir);
+    let files = await fs.readdir(dir);
+
+    // recursively validate files within sub-folders
+    for (const file of files) {
+      const filePathWithPrefix = `${dir}/${file}`;
+
+      // eslint-disable-next-line security/detect-non-literal-fs-filename
+      if (fs.lstatSync(filePathWithPrefix).isDirectory()) {
+        let newFilesInDirectory = await validatePathName(`${filePathWithPrefix}`);
```

Comment:
It seems like a minor point, but I really prefer to use `const` wherever possible. It feels especially not intuitive to be modifying the same variable that is being used in the loop on line 11. Could you refactor this code to use constants and could you introduce a new array instead of `files` that holds the returned files. Perhaps the new value holding all files could be `allFilesRecursive` or something like that

Reply:
Addressed this with a newer / cleaner way to handle all the paths. Let me know what you think.

Reply:
The new approach looks great! I'll merge and release a new version